test/nlp/simple-model.jl

```diff
@@ -32,6 +32,22 @@ function SimpleNLPModel()
   return SimpleNLPModel(meta, Counters())
 end
 
+function SimpleNLPModel32()
```

Comment:
You can define the constructor above as
```
function SimpleNLPModel(::Type{T}) where T
```
and another constructor `SimpleNLPModel() = SimpleNLPModel(Float64)`.